src/common/consensus-provider/consensus-provider.service.ts

```diff
@@ -290,10 +290,19 @@ export class ConsensusProviderService {
       return cached.missed ? undefined : cached.info;
     }
 
+    const workingMode = this.config.get('WORKING_MODE');
+
     const blockInfo = await this.retryRequest<BlockInfoResponse>(
       async (apiURL: string) => this.apiGet(apiURL, this.endpoints.blockInfo(blockId)),
       {
         maxRetries: this.config.get('CL_API_GET_BLOCK_INFO_MAX_RETRIES'),
+        useFallbackOnResolved: (r) => {
+          if (r.hasOwnProperty('finalized') && !r.finalized && workingMode === WorkingMode.Finalized) {
```

Comment:
rearrange the condition in order from most frequent (or more general) to less.
e.g. check the working mode and then the rest conditions

Reply:
OK, I don't mind. Changed the order of these checkings so that the most general ones now come first.